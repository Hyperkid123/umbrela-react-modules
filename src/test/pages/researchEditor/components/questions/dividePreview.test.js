import React from 'react'
import DividePreview from '../../../../../pages/researchEditorModule/components/questions/dividePreview';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Divide Preview component', () => {
  it('should render correctly empty', () => {
    const tree = shallow(
      <DividePreview
        options={[]}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render correctly with options', () => {
    const tree = shallow(
      <DividePreview
        options={[{
          optionId: 1,
          title: 'foo'
        }, {
          optionId: 2,
          title: 'foo2'
        }]}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should update points', () => {
    const wrapper = shallow(
      <DividePreview
        options={[{
          optionId: 1,
          title: 'foo'
        }, {
          optionId: 2,
          title: 'foo2'
        }]}
        scalePoints={100}
      />
    );
    wrapper.instance().handleDivideValue(0, 50);
    expect(wrapper.instance().remainingPoints()).toEqual(50);
    wrapper.instance().handleDivideValue(1, 100);
    expect(wrapper.instance().remainingPoints()).toEqual(50);
  });
});
