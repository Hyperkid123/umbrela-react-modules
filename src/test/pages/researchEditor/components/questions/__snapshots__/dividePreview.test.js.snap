// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divide Preview component should render correctly empty 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <label>
    : 
    NaN
  </label>
</div>
`;

exports[`Divide Preview component should render correctly with options 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <label>
    : 
    NaN
  </label>
  <styled.div
    key="1"
  >
    <TextField
      min={0}
      name="divide_input_1"
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      select={false}
      style={
        Object {
          "flexShrink": 0,
          "width": 150,
        }
      }
      type="number"
      value={0}
    />
    <styled.div
      column={true}
      grow={true}
      style={
        Object {
          "marginLeft": 10,
          "paddingTop": 10,
        }
      }
    >
      foo
      <WithStyles(LinearProgress)
        mode="determinate"
        value={NaN}
      />
    </styled.div>
  </styled.div>
  <styled.div
    key="2"
  >
    <TextField
      min={0}
      name="divide_input_2"
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      select={false}
      style={
        Object {
          "flexShrink": 0,
          "width": 150,
        }
      }
      type="number"
      value={0}
    />
    <styled.div
      column={true}
      grow={true}
      style={
        Object {
          "marginLeft": 10,
          "paddingTop": 10,
        }
      }
    >
      foo2
      <WithStyles(LinearProgress)
        mode="determinate"
        value={NaN}
      />
    </styled.div>
  </styled.div>
</div>
`;
